plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -640,6 +645,28 @@ else if (value instanceof Document) {
                 typeSignature = new TypeSignature(StandardTypes.ROW, parameters);
             }
         }
+        else if (value instanceof DBRef) {
+            List<TypeSignatureParameter> parameters = new ArrayList<>();
+
+            Optional<TypeSignature> idFieldType = guessFieldType(((DBRef) value).getId());
+            if (idFieldType.isPresent()) {
+                parameters.add(TypeSignatureParameter.namedTypeParameter(new NamedTypeSignature(Optional.of(new RowFieldName(ID)), idFieldType.get())));
+            }
+
+            Optional<TypeSignature> dbNameFieldType = ((DBRef) value).getDatabaseName() != null ? Optional.ofNullable(createUnboundedVarcharType().getTypeSignature()) : Optional.empty();
+            if (dbNameFieldType.isPresent()) {
+                parameters.add(TypeSignatureParameter.namedTypeParameter(new NamedTypeSignature(Optional.of(new RowFieldName(DATABASE_NAME)), dbNameFieldType.get())));
+            }
+
+            Optional<TypeSignature> collectionNameFieldType = ((DBRef) value).getCollectionName() != null ? Optional.ofNullable(createUnboundedVarcharType().getTypeSignature()) : Optional.empty();
+            if (collectionNameFieldType.isPresent()) {
+                parameters.add(TypeSignatureParameter.namedTypeParameter(new NamedTypeSignature(Optional.of(new RowFieldName(COLLECTION_NAME)), collectionNameFieldType.get())));
+            }
+
+            if (!parameters.isEmpty()) {
+                typeSignature = new TypeSignature(StandardTypes.ROW, parameters);
+            }
```

Comment:
I would simplify like below
```java
            parameters.add(TypeSignatureParameter.namedTypeParameter(new NamedTypeSignature(Optional.of(new RowFieldName(DATABASE_NAME)), VARCHAR.getTypeSignature())));
            parameters.add(TypeSignatureParameter.namedTypeParameter(new NamedTypeSignature(Optional.of(new RowFieldName(COLLECTION_NAME)), VARCHAR.getTypeSignature())));

            typeSignature = new TypeSignature(StandardTypes.ROW, parameters);
```

Reply:
Have handled this in my fork. Will commit.